subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/NoHistoryArtifactState.java

```diff
@@ -42,7 +42,7 @@ public boolean isAllowedToUseCachedResults() {
     }
 
     @Override
-    public BuildCacheKey calculateCacheKey() {
+    public TaskExecution getCurrentExecution() {
         return null;
```

Comment:
(In combination with what I suggested above that we should be returning a `BuildCacheKey` here).  We should make it so we never return a null `BuildCacheKey`.  We should have a "no cache key" value that we use instead.